functions/src/create-contract.ts

```diff
@@ -26,66 +27,42 @@ import {
   getFreeAnswerAnte,
   getNumericAnte,
   HOUSE_LIQUIDITY_PROVIDER_ID,
-  MINIMUM_ANTE,
 } from '../../common/antes'
 import { getNoneAnswer } from '../../common/answer'
 import { getNewContract } from '../../common/new-contract'
 import { NUMERIC_BUCKET_COUNT } from '../../common/numeric-constants'
 
-export const createContract = newEndpoint(['POST'], async (req, _res) => {
-  const [creator, _privateUser] = await lookupUser(await parseCredentials(req))
-  let {
-    question,
-    outcomeType,
-    description,
-    initialProb,
-    closeTime,
-    tags,
-    min,
-    max,
-    manaLimitPerUser,
-  } = req.body || {}
-
-  if (!question || typeof question != 'string')
-    throw new APIError(400, 'Missing or invalid question field')
-
-  question = question.slice(0, MAX_QUESTION_LENGTH)
-
-  if (typeof description !== 'string')
-    throw new APIError(400, 'Invalid description field')
-
-  description = description.slice(0, MAX_DESCRIPTION_LENGTH)
-
-  if (tags !== undefined && !Array.isArray(tags))
-    throw new APIError(400, 'Invalid tags field')
-
-  tags = (tags || []).map((tag: string) =>
-    tag.toString().slice(0, MAX_TAG_LENGTH)
-  )
+const bodySchema = z.object({
+  question: z.string().min(1).max(MAX_QUESTION_LENGTH),
+  description: z.string().max(MAX_DESCRIPTION_LENGTH),
+  tags: z.array(z.string().min(1).max(MAX_TAG_LENGTH)).optional(),
+  closeTime: z.number(),
```

Comment:
Check if closeTime is positive, in the future, and not too far in the future (< 100 years — or whatever you think the lifetime of Manifold is).

Reply:
👍